### ISSUE TITLE
Unable to craft "Flint and Steel" in Survival Mode

### ISSUE THREAD

**Issue by rekuen** [cuberite/cuberite#2722]
Hi, I've found that when you try to craft a "Flint and Steel" or lighter, in survival mode, it just doesn't work, you put the pieces in the workbench, the lighter appears for a sec in the result and then dissapears again, if you click in the empty result the lighter appears again but when you add it to your invetory and leave the workbench, the lighter is vanished from the inventory and the materials used to crafted it are thrown from workbench like when you don't finish crafting and item completely.

Is the exact same issue that happend with the "Lead" item a while ago, but seems to be fixed with the Lead.

I can make a video to make it clear if you need it.  I'm unning the latest build from http://builds.cuberite.org/ .

**Comment by SphinxC0re** [comment]
A video would be overkill. A gif would be nice though

**Comment by NiLSPACE** [comment]
Perhaps you removed the `crafting.txt`? Or did you change the `items.ini` file? Maybe you changed the `items.ini` file to the [German one](https://github.com/cuberite/cuberite/blob/master/Server/lang/items_de.ini)? Would crafting still work with that?

**Comment by SphinxC0re** [comment]
It should

**Comment by rekuen** [comment]
I have all the files intact, just unzip the latest build yesterday night, it wasn't working before (same thing happend with the Lead item) I'm making a video right now.

**Comment by rekuen** [comment]
Demo video, sorry for the quality:
https://youtu.be/k9PCsXnDdqI
This time the lighter is crafted fine but after a while disappears again, some times it just disappears when you add it to the inventory.

**Comment by NiLSPACE** [comment]
If I read the [crafting file](https://github.com/cuberite/cuberite/blob/master/Server/crafting.txt#L195-L196) correctly you can only craft a lighter by placing the items like this:
```
X: Flint
Y: Iron
O: Nothing
   OX
   YO
or
   XO
   OY
not like this:
XY
```

This is pretty easy to fix.

**Comment by rekuen** [comment]
You are right, if I try that combination it doesn't disappear. I'm pretty new to minecraft and I try the combinations i see online when i don't know how to craft something, YX is what I found online and doesn't work also. I will look into crafting.txt next time. 

Thanks.

**Comment by rekuen** [comment]
The flickering sec when the lighter shows briefly is what make me think i was positioning right but it wasn't working.

**Comment by NiLSPACE** [comment]
If we change the lines in the crafting file to
```
Lighter = IronIngot, * | Flint, *
```
it should work. Perhaps you can try it. Just change line 195 and 196 with this.

**Comment by rekuen** [comment]
I can confirm that solves the problem, just tried in my server and is working.

**Comment by NiLSPACE** [comment]
The lead can probably be fixed in the same way. Perhaps you can try :) If not, could you give me a link to the proper crafting recipe?

**Comment by rekuen** [comment]
The lead is fixed as far as I know, I tried this morning, the thing that is different from vanilla, is that you get one lead instead of 2. 

http://www.minecraftinfo.com/lead.htm
http://minecraft.gamepedia.com/Lead

**Comment by NiLSPACE** [comment]
Oh, that's easy. Change [this (line 206)](https://github.com/cuberite/cuberite/blob/master/Server/crafting.txt#L206) to this:
```
Lead, 2 = String, 1:1, 1:2, 2:1, 3:3 | Slimeball, 2:2
```

**Comment by rekuen** [comment]
Working perfect, thanks. 
Now that I know how this works I will fixed future items that i may find with errors and make a pull request.

Thanks again.

**Comment by madmaxoft** [comment]
Flint and steel recipe has been changed to a shapeless one in 1.7.2, according to the wiki:
http://minecraft.gamepedia.com/Flint_%26_Steel#History
I had no idea you could craft it in any other way :)

**Comment by NiLSPACE** [comment]
I didn't notice as well. Sneaky Mojang is sneaky O_O

**Comment by SafwatHalaby** [comment]
Fixed in #2725
